tedana/decomposition/ica.py

```diff
@@ -69,18 +233,19 @@ def tedica(data, n_components, fixed_seed, maxit=500, maxrestart=10):
             w = list(filter(lambda i: issubclass(i.category, UserWarning), w))
             if len(w):
                 LGR.warning(
-                    f"ICA with random seed {fixed_seed} failed to converge after {ica.n_iter_} "
-                    "iterations"
+                    "ICA with random seed {0} failed to converge after {1} "
+                    "iterations".format(fixed_seed, ica.n_iter_)
                 )
                 if i_attempt < maxrestart - 1:
                     fixed_seed += 1
-                    LGR.warning(f"Random seed updated to {fixed_seed}")
+                    LGR.warning("Random seed updated to {0}".format(fixed_seed))
             else:
                 LGR.info(
-                    f"ICA with random seed {fixed_seed} converged in {ica.n_iter_} iterations"
+                    "ICA with random seed {0} converged in {1} "
+                    "iterations".format(fixed_seed, ica.n_iter_)
                 )
                 break
 
     mmix = ica.mixing_
     mmix = stats.zscore(mmix, axis=0)
-    return mmix, fixed_seed
+    return mmix, fixed_seed
```

Comment:
GitHub shows a red circled dash at the end of some files. This indicates that the file does not contain a newline at the end. The final non-empty line of the file should contain a newline, but there should be no empty lines beyond that.